### ISSUE TITLE
backtesting in the MT5 is not working

### ISSUE THREAD

**Issue by mokrslar** [khramkov/MQL5-JSON-API#13]
First of all thank you for your great work. 
I have noticed, that expert advisor cannot be backtestet in MT5 strategy tester. The reason seems to be that strategy tester cannot communicate through sockets. Is there some workaround, am I missing something? I found a potential workaround, but it is for MT4, and at the time do not have the resources and knowledge to implement it - details in the link. Thank you!

http://www.fxdesignlab.com/wp-content/uploads/2020/08/EA-Comm-by-Sockets-with-the-MT4-Tester.pdf

**Comment by device-host** [comment]
The whole point of this advisor is to give external access to MetaTrader - not to run as a strategy... It is not a real expert advisor.
See the python back-testing implementations that use this interface like https://github.com/khramkov/Backtrader-MQL5-API

**Comment by oswaldmotape** [comment]
> The whole point of this advisor is to give external access to MetaTrader - not to run as a strategy... It is not a real expert advisor.
> See the python back-testing implementations that use this interface like https://github.com/khramkov/Backtrader-MQL5-API

Correct, access to the data and use the data to create strategies and send orders to your favourite broker API if they don't provide raw data

**Comment by PolaNaguib** [comment]
Can someone help with this? https://www.mql5.com/en/forum/445521  Thanks
